src/channel_state.js

```diff
@@ -251,23 +251,43 @@ export class ChannelState {
 	 * @return {boolean} Returns if the message was removed
 	 */
 	removeMessage(messageToRemove) {
-		let removed = false;
-		const messages = this.messages.flatMap(message => {
```

Comment:
We didn't handle the case of threads here. Because of which "cancel" action on giphy command didn't work in our web and mobile sdk.